src/ConfigHandler.cpp

```diff
@@ -10,9 +10,29 @@
 
 using namespace std;
 
-void ConfigHandler::loadConfig(string filename) {
-  cout << "loadConfig() in ConfigHandler" << endl;
+/*  Initialize static member */
+shared_ptr<ConfigHandler> ConfigHandler::mInstance{nullptr};
+
+ConfigHandler& ConfigHandler::GetInstance() {
+  if (mInstance == nullptr)
+    mInstance = shared_ptr<ConfigHandler>(new ConfigHandler);
```

Comment:
fyi
https://stackoverflow.com/a/18301738

Reply:
`mInstance = make_shared<ConfigHandler>();`
won't compile, since the default constructor is private... just found that out when trying to do so... :thinking:

Reply:
haha nice^^
I encountered that problem before, but I dont know if i like the [workaround](https://stackoverflow.com/a/8147213)

it's up to you